app/src/main/java/com/norvellium/tasky/core/validation/ValidateEmail.kt

```diff
@@ -0,0 +1,20 @@
+package com.norvellium.tasky.core.validation
+
+import android.content.Context
+import androidx.core.util.PatternsCompat
+import com.norvellium.tasky.R
+
+class ValidateEmail(context: Context) {
+
+    private val resources = context.resources
+
+    fun validate(email: String): ValidationResult {
+        if (email.isBlank()) {
+            return ValidationResult(false, resources.getString(R.string.validation_error_email_blank))
+        }
+        if (!PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()) {
+            return ValidationResult(false, resources?.getString(R.string.validation_error_email_invalid))
+        }
+        return ValidationResult(true)
+    }
+}
```

Comment:
Validating an email is business logic and ideally you'd want business logic to be testable with a local unit test (not inside androidTest) which doesn't work if you pass the context or use Android specific dependencies (such as `PatternsCompat`).
In this class there's not too much to test other than returning the right result if the email is blank, but generally I'd try to isolate such business logic as much as possible. 

Every time, you have implementation details and want to get rid of that, the procedure is the same:

1. You create an abstraction (interface)
2. You write an implementation of that (in this case using Patterns.EMAIL_ADDRESS...)
3. You pass the abstraction to the constructor of your class needing i
4. Now you can decide which implementation you pass when you create that class/use case
For example:

```
interface EmailPatternValidator {
    fun isValidEmailPattern(email: String): Boolean
}
```

```
class EmailPatternValidatorImpl: EmailPatternValidator {

    override fun isValidEmailPattern(email: String): Boolean {
        return PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Then inject an `EmailPatternValidator` into your `ValidateEmail` classes constructor and use there.

To get rid of the context, you could return result codes (in form of an enum class for example: `ValidationResult.EMAIL_BLANK`) and then map these to string resource IDs in your ViewModel.